src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -104,9 +108,9 @@ const VersionsItem = ({
 
     // Version action helpers
     const canPreview = can_preview && !isDeleted && !isLimited && !isRestricted;
-    const showDelete = can_delete && !isDeleted && !isCurrent;
+    const showDelete = can_delete && !isDeleted && !shouldDisablePromoteAndDelete && !isCurrent;
     const showDownload = can_download && !isDeleted && isDownloadable;
-    const showPromote = can_upload && !isDeleted && !isCurrent;
+    const showPromote = can_upload && !isDeleted && !shouldDisablePromoteAndDelete && !isCurrent;
     const showRestore = can_delete && isDeleted;
```

Comment:
Should we also not allow restoring deleted versions?

Reply:
They're restored in archiving process, but maybe this could be disabled just in case too